stepler/nova/tests/test_metadata.py

```diff
@@ -35,6 +35,7 @@ def test_metadata_reach_all_booted_vm(security_group, nova_floating_ip,
                                             network, keypair, [security_group])
 
         server_steps.attach_floating_ip(server, nova_floating_ip)
+        server.get()
```

Comment:
Ok, lets look deeply to chunk:

```python
server_steps.attach_floating_ip(server, nova_floating_ip)
server_steps.check_ssh_connect(server, keypair, username='ubuntu', timeout=600)

@step
def attach_floating_ip(self, server, floating_ip, check=True):
    """Step to attach floating IP to server."""
    self._client.add_floating_ip(server, floating_ip)
  
    if check:
        server.get()
        floating_ips = self.get_ips(server, 'floating').keys()

        assert_that(floating_ips, has_items(floating_ip.ip), (
            "Floating IP not in a list of server's IPs."))

```

You call `attach_floating_ip` with `check=True` by default, it means that step will be verified and object `server` will be update. And the `server` will be passed to `check_ssh_connect`.